orderer/common/cluster/rpc.go

```diff
@@ -154,6 +155,8 @@ func (s *RPC) getOrCreateStream(destination uint64, operationType OperationType)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	stub.SourceNodeID = s.SourceNodeID
```

Comment:
you don't need to do this. The `Stub` struct contains the `RemoteNode` which has the `ID` you need.

Reply:
Stub does not hold the origin/source node id. It currently contains only the destination node id.